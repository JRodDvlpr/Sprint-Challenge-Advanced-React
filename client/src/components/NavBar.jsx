import React from 'react';
import {useDarkMode} from '../hooks/useDarkMode';

const Navbar = () => {
  // grabbing from my darkmode component
  const [darkMode, setDarkMode] = useDarkMode(false);
  const toggleMode = e => {
    e.preventDefault();
    setDarkMode(!darkMode);
  };
  return (
    <nav className="navbar">
      <h1>Women World Cup</h1>
      <div className="dark-mode__toggle">
        <button
          onClick={toggleMode}
          className={darkMode ? 'toggle toggled' : 'toggle'}>
        </button>
      </div>
    </nav>
  );
};

export default Navbar;