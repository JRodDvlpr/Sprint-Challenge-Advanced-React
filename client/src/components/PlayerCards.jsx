import React from 'react';

const PlayerCards = (props) => {

    return(
        <div>
            <h2>{props.name}</h2>
            <p>{props.country}</p>
            <p>{props.searches}</p>
        </div>
    )
}

export default PlayerCards;