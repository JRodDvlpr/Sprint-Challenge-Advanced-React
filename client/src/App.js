import React from 'react';
import './App.css';
import NavBar from './components/NavBar';
import axios from 'axios';
import PlayerCards from './components/PlayerCards';

class App extends React.Component {
  constructor() {
    super();
    this.state = {
      players: []
    };
  }
  componentDidMount() {
    axios.get("http://localhost:5000/api/players")
      .then(res => {
        console.log(res.data);
        this.setState({ players: res.data });
      })
      .catch(error => console.log(error));
  }

  render() {
    return (
      <div className="App">
       <div>
        <NavBar />
       </div>
       <div data-testid="player-list">
         {this.state.players.map(data =>(
           <PlayerCards id={data.id} key={data.id} country={data.country} searches={data.searches} name={data.name} />
         ))}
       </div>
      </div>
    );
  }
}


export default App;
