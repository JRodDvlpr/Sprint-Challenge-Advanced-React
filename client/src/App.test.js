import React from 'react';
import ReactDOM from 'react-dom';
import App from './App';
import * as rtl from '@testing-library/react';
import NavBar from './components/NavBar';
import * as Jest from '@testing-library/jest-dom/extend-expect';
import PlayerCards from './components/PlayerCards';
import renderer from 'react-test-renderer';

it('renders without crashing', () => {
  const div = document.createElement('div');
  ReactDOM.render(<App />, div);
  ReactDOM.unmountComponentAtNode(div);
});

test('renders app without crashing', () => {
  rtl.render(<App />);
});

test('player list', ()=>{
  const {getByTestId} = rtl.render (<NavBar />);
  expect(getByTestId("dark-mode")).toHaveClass("dark-mode__toggle")
});

test('reflects World Cup', () =>{
  const {getAllByText} = rtl.render (<NavBar />);
  getAllByText(/World/i);
  getAllByText(/Cup/i);
});

it('renders Player Card', () => {
  const tree = renderer.create(<PlayerCards />).toJSON();
  expect(tree).toMatchSnapshot();
});