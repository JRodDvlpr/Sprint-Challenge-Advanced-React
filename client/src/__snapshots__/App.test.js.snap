// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders Player Card 1`] = `
<div>
  <h2 />
  <p />
  <p />
</div>
`;
